### PR TITLE
[train] enable new persistence mode for core and serve tests

### PR DESCRIPTION
**Pull request by matthewdeng**
## Why are these changes needed?

This enables the new persistence mode for the following tests:

| Test | Root Cause |
|--------|--------|
| :cold_face: :python: :serverless: Python 3.7 Serve Tests | `test_air_integrations.py` |
| :ray-serve: serve: serve tests | `test_air_integrations.py` |
| :ray: core: python tests | `test_multi_node_3.py`, `test_usage_stats.py` |
| :tv: :serverless: Serve Tests | `test_air_integrations_gpu.py` | 
| :redis: (External Redis) (Large) | Flaky | 
| :serverless: Serve Tests (streaming FF off) | `test_air_integrations.py` | 
| :serverless: Serve Tests (streaming and routing FFs off) | `test_air_integrations.py` | 
| :python: (ASAN tests) | Flaky | 
| :kubernetes: :mending_heart: :ray-serve: serve chaos network delay test | Flaky | 

I originally also included `:windows: Build & Test`, but there is a failure in `test_client_library_integration` that might require code changes: 
<details><summary><b>Stacktrace</b></summary>
<p>

```
_bk;t=1693015964187================================== FAILURES ===================================
_bk;t=1693015964187_________________________ test_rllib_integration_tune _________________________
_bk;t=1693015964187
_bk;t=1693015964187ray_start_regular = RayContext(dashboard_url='127.0.0.1:8265', python_version='3.8.10', ray_version='3.0.0.dev0', ray_commit='{{RAY_COMMIT_SHA}}', protocol_version=None)
_bk;t=1693015964187
_bk;t=1693015964187    def test_rllib_integration_tune(ray_start_regular):
_bk;t=1693015964187        with ray_start_client_server():
_bk;t=1693015964187            # Confirming the behavior of this context manager.
_bk;t=1693015964187            # (Client mode hook not yet enabled.)
_bk;t=1693015964187            assert not client_mode_should_convert()
_bk;t=1693015964187            # Need to enable this for client APIs to be used.
_bk;t=1693015964187            with enable_client_mode():
_bk;t=1693015964187                # Confirming mode hook is enabled.
_bk;t=1693015964187                assert client_mode_should_convert()
_bk;t=1693015964187>               tune.run(
_bk;t=1693015964187                    "DQN", config={"env": "CartPole-v1"}, stop={"training_iteration": 2}
_bk;t=1693015964187                )
_bk;t=1693015964187
_bk;t=1693015964187\\?\C:\Users\ContainerAdministrator\AppData\Local\Temp\Bazel.runfiles_qgfvjrfp\runfiles\com_github_ray_project_ray\python\ray\tests\test_client_library_integration.py:46: 
_bk;t=1693015964187_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
_bk;t=1693015964187c:\install\ray\python\ray\tune\tune.py:608: in run
_bk;t=1693015964187    return ray.get(remote_future)
_bk;t=1693015964187c:\install\ray\python\ray\_private\auto_init_hook.py:24: in auto_init_wrapper
_bk;t=1693015964187    return fn(*args, **kwargs)
_bk;t=1693015964187c:\install\ray\python\ray\_private\auto_init_hook.py:24: in auto_init_wrapper
_bk;t=1693015964187    return fn(*args, **kwargs)
_bk;t=1693015964187c:\install\ray\python\ray\_private\client_mode_hook.py:102: in wrapper
_bk;t=1693015964187    return getattr(ray, func.__name__)(*args, **kwargs)
_bk;t=1693015964187c:\install\ray\python\ray\util\client\api.py:42: in get
_bk;t=1693015964187    return self.worker.get(vals, timeout=timeout)
_bk;t=1693015964187c:\install\ray\python\ray\util\client\worker.py:434: in get
_bk;t=1693015964187    res = self._get(to_get, op_timeout)
_bk;t=1693015964187_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
_bk;t=1693015964187
_bk;t=1693015964187self = <ray.util.client.worker.Worker object at 0x000001B3FF148130>
_bk;t=1693015964187ref = [ClientObjectRef(ff82f6faafdd0e77ffffffffffffffffffffffff0100000001000000)]
_bk;t=1693015964187timeout = 2.0
_bk;t=1693015964187
_bk;t=1693015964187    def _get(self, ref: List[ClientObjectRef], timeout: float):
_bk;t=1693015964187        req = ray_client_pb2.GetRequest(ids=[r.id for r in ref], timeout=timeout)
_bk;t=1693015964187        data = bytearray()
_bk;t=1693015964187        try:
_bk;t=1693015964187            resp = self._get_object_iterator(req, metadata=self.metadata)
_bk;t=1693015964187            for chunk in resp:
_bk;t=1693015964187                if not chunk.valid:
_bk;t=1693015964187                    try:
_bk;t=1693015964187                        err = cloudpickle.loads(chunk.error)
_bk;t=1693015964187                    except (pickle.UnpicklingError, TypeError):
_bk;t=1693015964187                        logger.exception("Failed to deserialize {}".format(chunk.error))
_bk;t=1693015964187                        raise
_bk;t=1693015964187>                   raise err
_bk;t=1693015964187E                   ray.exceptions.RayTaskError(AssertionError): [36mray::run()[39m (pid=3876, ip=172.17.161.116)
_bk;t=1693015964187E                     File "python\ray\_raylet.pyx", line 1613, in ray._raylet.execute_task
_bk;t=1693015964187E                     File "c:\install\ray\python\ray\tune\tune.py", line 852, in run
_bk;t=1693015964187E                       experiments[i] = Experiment(
_bk;t=1693015964187E                     File "c:\install\ray\python\ray\tune\experiment\experiment.py", line 204, in __init__
_bk;t=1693015964187E                       self.storage = StorageContext(
_bk;t=1693015964187E                     File "c:\install\ray\python\ray\train\_internal\storage.py", line 486, in __init__
_bk;t=1693015964187E                       self._check_validation_file()
_bk;t=1693015964187E                     File "c:\install\ray\python\ray\train\_internal\storage.py", line 514, in _check_validation_file
_bk;t=1693015964187E                       if not _exists_at_fs_path(fs=self.storage_filesystem, fs_path=valid_file):
_bk;t=1693015964187E                     File "c:\install\ray\python\ray\train\_internal\storage.py", line 255, in _exists_at_fs_path
_bk;t=1693015964187E                       assert not is_uri(fs_path), fs_path
_bk;t=1693015964187E                   AssertionError: C:/tmp/4lhdprva/execroot/com_github_ray_project_ray/_tmp/76795314268c91c8b7ca920989e00292\DQN_2023-08-26_02-12-13\.validate_storage_marker

```

</p>
</details> 

## Checks

See https://buildkite.com/ray-project/oss-ci-build-pr/builds/33892

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/tests/test_usage_stats.py [hunk 1]
```diff
@@ -1206,9 +1206,7 @@ def run_usage_stats_server(reporter):
         if os.environ.get("RAY_MINIMAL") != "1":
             expected_payload["tune_scheduler"] = "FIFOScheduler"
             expected_payload["tune_searcher"] = "BasicVariantGenerator"
-            expected_payload["air_storage_configuration"] = "driver"
             expected_payload["air_entrypoint"] = "Tuner.fit"
```
Comment (woshiyyya) [inline python/ray/tests/test_usage_stats.py:1209]: What is this flag used for?

**Review by woshiyyya — APPROVED** [review]
Lgtm.

File: python/ray/tests/test_usage_stats.py (hunk 1 above)
Comment (matthewdeng) [reply python/ray/tests/test_usage_stats.py:1209]: This is for telemetry, but we disabled syncing to driver.

**Review by justinvyu — APPROVED** [review]
thanks 💯

**Pull request merged**